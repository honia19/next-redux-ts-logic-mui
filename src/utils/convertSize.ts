export enum Unit {
  Bytes = 'B',
  Kilobytes = 'KB',
  Megabytes = 'MB',
  Gigabytes = 'GB',
  Terabytes = 'TB',
}

const convertSize = (bytes: number, unit: Unit): string => {
  const units = Object.values(Unit);
  const unitIndex = units.indexOf(unit);
  const size = bytes / Math.pow(1024, unitIndex);

  return size.toFixed(1) + ' ' + unit;
};

export default convertSize;
