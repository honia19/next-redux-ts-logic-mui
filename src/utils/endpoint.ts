const endpoint = (
  feature: string,
  method: string,
  route: string,
): { endpoint: string; url: string } => ({
  endpoint: `${feature} ${method} ${route}`,
  url: route,
});

export default endpoint;
