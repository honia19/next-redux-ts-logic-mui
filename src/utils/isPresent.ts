import { isNil, isEmpty } from 'ramda';

const isPresent = <T>(data: T): boolean => !isNil(data) && !isEmpty(data);

export default isPresent;
