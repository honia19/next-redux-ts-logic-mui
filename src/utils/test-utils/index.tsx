import { ThemeProvider } from '@mui/material';
import { render } from '@testing-library/react';
import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { theme } from '@/theme';

import { RenderProviderType } from './types';

const mockStore = configureStore([]);
const initialState = {};
const store = mockStore(initialState);

const ReduxProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>{children}</Provider>
    </ThemeProvider>
  );
};

const MAP_PROVIDERS = {
  redux: ReduxProvider,
};

const customRender = (
  ui: ReactElement,
  options?: { wrapperType?: RenderProviderType },
) => {
  const { wrapperType = 'redux', ...restOptions } = options || {};
  return render(ui, { wrapper: MAP_PROVIDERS[wrapperType], ...restOptions });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';
// eslint-disable-next-line import/export
export { customRender as render };
