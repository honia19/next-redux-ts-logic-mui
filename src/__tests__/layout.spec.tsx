import React from 'react';

import { render, screen } from '@/utils/test-utils';

import RootLayout, { metadata } from '../app/layout';

describe('Layout render', () => {
  it('should render correctly', () => {
    render(<RootLayout>Hello, World!</RootLayout>);

    screen.getByText('Hello, World!');
  });
});

describe('Metadata', () => {
  it('should render correctly', () => {
    expect(metadata).toMatchSnapshot();
  });
});
