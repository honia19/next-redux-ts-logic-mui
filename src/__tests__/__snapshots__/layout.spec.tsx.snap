// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metadata should render correctly 1`] = `
{
  "description": "Portfolio for interview task",
  "icons": [
    {
      "rel": "icon",
      "url": "/img.jpg",
    },
    {
      "rel": "icon",
      "sizes": "32x32",
      "type": "image/png",
      "url": "/img.jpg",
    },
    {
      "rel": "icon",
      "sizes": "16x16",
      "type": "image/png",
      "url": "/img.jpg",
    },
    {
      "rel": "apple-touch-icon",
      "sizes": "180x180",
      "url": "/img.jpg",
    },
    {
      "rel": "icon",
      "sizes": "192x192",
      "url": "/img.jpg",
    },
  ],
  "title": "Images app",
}
`;
